routes/Program/controllers.js

```diff
@@ -9,24 +9,140 @@ const router = express.Router();
 router.post('/', checkSchema(programSchema), async (req, res, next) => {
    const {
       name,
-      collegeId
+      collegeId,
+      degree,
+      language,
+      type,
+      accredited,
+      duration,
+      organization,
+      credits,
+      totalPlaces,
+      taxFreePlaces,
+      taxPlaces,
+      admissionType,
+      courses,
+      competencies
    } = req.body;
 
    try {
-      await service.create(name, ObjectId(collegeId));
+      await service.create(name, ObjectId(collegeId), degree, language, type, accredited, duration, organization, credits,
+          totalPlaces, taxFreePlaces, taxPlaces, admissionType, courses, competencies);
 
       res.status(201).end();
    } catch (err) {
       next(err);
    }
 });
 
-router.get('/', checkSchema(querySchema), async (req, res, next) => {
+router.get('/:id', checkSchema(querySchema), async (req, res, next) => {
+   const {
+      id
+   } = req.params;
+
    try {
-      res.status(200).json(await service.getAll(req.query));
+      res.status(200).json(await service.getById(id, querySchema));
    } catch (err) {
       next(err);
    }
 });
 
+router.get('', checkSchema(querySchema), async (req, res, next) => {
```

Comment:
I think most of those conditions could be refactored in something simpler. For example, you could refactor the service methods such as `getByDegree`, `getByLanguage` etc, in a single `getFiltered` and pass the entire `req.query` object. This will reduce all this cases to a single one.

In getFiltered method you can simply filter the query for values outside of null/undefined and build a simple condition from those, such as
```javascript
Program.find({
      degree: degree,
      language: language,
      ...
   })
 ```
In this case, we really need pagination, which should be found in the query under the identifiers declared in querySchema. So add them with skip and limit, but take care to not end in the find conditions as well.

Maybe some default pagination parameters would be good as well, i case we somehow forget to send them from the frontend.

Reply:
done